mixnet/util/src/lib.rs

```diff
@@ -0,0 +1,42 @@
+use std::{collections::HashMap, net::SocketAddr, sync::Arc};
+
+use parking_lot::Mutex;
+use tokio::net::TcpStream;
+
+#[derive(Clone)]
+#[repr(transparent)]
+pub struct ConnectionPool {
+    pool: Arc<Mutex<HashMap<SocketAddr, Arc<tokio::sync::Mutex<TcpStream>>>>>,
```

Comment:
how's keepalive handled by the pool?

Reply:
Yeah, currently we do not check if the conn is still living. As @youngjoon-lee mentioned, the implementation is not fail-safe. I will implement dead connection checks and cleanup in the next PR.

Reply:
Although this can be done in the next PR, I guess the easiest way might be the "on-demand reconnection", instead of checking the status of connections in the pool periodically.